### PR TITLE
Teslium and liquid dark matter scaling

### PR DESCRIPTION
**Pull request by lass9030**
As title says, adds scaling to Teslium and Liquid dark matter.
Whether or not this is a good scale, i will let you decide.

:cl: Alonefromhell
balance: Teslium and Liquid Dark Matter should scale better now.
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/recipes/pyrotechnics.dm [hunk 1]
```diff
@@ -121,10 +121,10 @@
 
 /datum/chemical_reaction/liquid_dark_matter/on_reaction(datum/reagents/holder, created_volume)
 	if(holder.has_reagent("stabilizing_agent"))
-		return
-	holder.remove_reagent("liquid_dark_matter", created_volume)
+		return FALSE
```
Comment (SamHPurp) [inline code/modules/reagents/chemistry/recipes/pyrotechnics.dm]: why returning false? I don't think it takes a return value

Comment (Crazylemon64) [inline code/modules/reagents/chemistry/recipes/pyrotechnics.dm]: The return value of `on_reaction` is never used - this value should be omitted.

File: code/modules/reagents/chemistry/recipes/pyrotechnics.dm [hunk 2]
```diff
@@ -110,7 +110,7 @@
 
 /datum/chemical_reaction/sorium_vortex/on_reaction(datum/reagents/holder, created_volume)
 	var/turf/simulated/T = get_turf(holder.my_atom)
-	goonchem_vortex(T, 1, 5, 6)
+	goonchem_vortex(T, 1, round(created_volume / 33), round(created_volume / 25))
```
Comment (Crazylemon64) [inline code/modules/reagents/chemistry/recipes/pyrotechnics.dm]: Example cases for balance design purposes:
To get 5 tiles of range, `created_volume = 33*5 = 165 units`
To get 6 steps of pull, `created_volume = 25*6 = 150 units`

(The same holds true for LDM)

File: code/modules/reagents/chemistry/recipes/pyrotechnics.dm [hunk 3]
```diff
@@ -411,9 +411,9 @@ datum/chemical_reaction/flash_powder
 
 /datum/chemical_reaction/shock_explosion/on_reaction(datum/reagents/holder, created_volume)
 	var/turf/T = get_turf(holder.my_atom)
-	for(var/mob/living/carbon/C in view(6, T))
+	for(var/mob/living/carbon/C in view(max(1, round(created_volume / 3)), T))
```
Comment (Crazylemon64) [inline code/modules/reagents/chemistry/recipes/pyrotechnics.dm]: Example case for teslium explosions:
To get the same range of effect:
`total reagents needed = (5+5)*(required created_volume) = 10*(3*6) = 180 units`

File: code/modules/reagents/chemistry/recipes/pyrotechnics.dm [hunk 4]
```diff
@@ -411,9 +411,9 @@ datum/chemical_reaction/flash_powder
...
 		C.Beam(T,icon_state="lightning[rand(1,12)]",icon='icons/effects/effects.dmi',time=5) //What? Why are we beaming from the mob to the turf? Turf to mob generates really odd results.
-		C.electrocute_act(3.5, "electrical blast")
+		C.electrocute_act(max(1, round(created_volume / 5)), "electrical blast")
```
Comment (Crazylemon64) [inline code/modules/reagents/chemistry/recipes/pyrotechnics.dm]: Example case for teslium explosions:
To get the same electrocution force:
`total reagents needed = (5+5)*(required created_volume) = 10*(3.5*5) = 175 units`

**Comment by Crazylemon64** [comment]
Per the freeze, please cite fixes for this balance PR.

**Comment by lass9030** [comment]
Forgot about the feature freeze. Seeing as i have no real fix PR's to cash in, I'll be closing this.
Thanks for all the feedback, though, much appreciated.

**Comment by farie82** [comment]
I'm willing to give some of my fix points for this because i think it's quite needed.
#9913 #9910 #9904

**Pull request closed without merging**